src/Components/Shared/src/ResourceCollectionProvider.cs

```diff
@@ -11,36 +11,29 @@ namespace Microsoft.AspNetCore.Components;
 
 internal class ResourceCollectionProvider
 {
-    private const string ResourceCollectionUrlKey = "__ResourceCollectionUrl";
     private string? _url;
+
+    [SupplyParameterFromPersistentComponentState]
+    public string? ResourceCollectionUrl
+    {
+        get => _url;
+        set => _url = value;
+    }
```

Comment:
```suggestion
    public string? ResourceCollectionUrl { get; set; }
```

Reply:
I would rather go towards removing `SetResourceCollectionUrl` and moving the exception logic into the setter. This can be done after the bug of preserving the values is found and fixed.